### ISSUE TITLE
bug: Rules fail to detect patterns with arrow functions and internal props destructuring

### ISSUE THREAD

**Issue by Segodnya** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#54]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.8.0

### ESLint version

9.34.0

### Describe the bug

The plugin fails to detect several rule violations when using arrow function components with props destructured inside the component body, rather than in the function parameters.

**Affected Rules:**
- `no-pass-live-state-to-parent`
- `no-pass-data-to-parent`
- `no-pass-ref-to-parent`

### Steps To Reproduce

**Examples that DON'T work:**

```tsx
// ❌ Not detected
const ChildWithLiveStateToParent = (props: ChildProps) => {
  const { onTextChanged } = props;
  const [text, setText] = useState('');

  useEffect(() => {
    onTextChanged(text);
  }, [onTextChanged, text]);

  return <input />;
};

// ❌ Not detected
interface RefProps {
  onRef: (ref: HTMLDivElement | null) => void;
}

function ChildWithRefToParent(props: RefProps) {
  const { onRef } = props;
  const ref = useRef<HTMLDivElement>(null);

  useEffect(() => {
    onRef(ref.current);
  }, [onRef, ref.current]);

  return <div ref={ref} />;
}
```

**Examples that DO work:**

```tsx
// ✅ Detected correctly
function ChildWithLiveStateToParent({ onTextChanged }) {
  const [text, setText] = useState('');

  useEffect(() => {
    onTextChanged(text);
  }, [onTextChanged, text]);

  return <input />;
}

// ✅ Detected correctly
function ChildWithRefToParent({ onRef }) {
  const ref = useRef(null);

  useEffect(() => {
    onRef(ref.current);
  }, [onRef, ref.current]);

  return <div ref={ref} />;
}
```

**Additional Issue with `no-pass-ref-to-parent`:**

The listener variant also fails when using intermediate variables or conditional checks:

```tsx
// ❌ Not detected
function ChildWithRefListener({ onClicked }) {
  const ref = useRef(null);

  useEffect(() => {
    const element = ref.current;  // Intermediate variable
    
    if (element) {  // Conditional check
      element.addEventListener('click', (event) => {
        onClicked(event);
      });
    }
  }, [onClicked]);
}

// ✅ Detected correctly
const ChildWithRefListener = ({ onClicked }) => {
  const ref = useRef(null);

  useEffect(() => {
    ref.current.addEventListener('click', (event) => {
      onClicked(event);
    });
  }, [onClicked]);
};
```

### Expected Behavior

The plugin should detect these patterns regardless of:

- Whether components use function declarations or arrow functions
- Whether props are destructured in parameters or inside the component body
- TypeScript type annotations on props
- Intermediate variables when tracing refs or callbacks
